lib/service/Service.ts

```diff
@@ -170,7 +170,32 @@ class Service extends EventEmitter {
     argChecks.HAS_HOST({ host });
     argChecks.VALID_PORT({ port });
 
+    const peerActive = new Promise((resolve) => {
+      this.pool.once('peer.active', (peerPubKey) => {
+        if (peerPubKey === nodePubKey) {
+          resolve();
+        }
+      });
+    });
+
+    const peerClose: Promise<string> = new Promise((resolve) => {
+      this.pool.once('peer.close', (peerPubKey, rejectionMsg) => {
+        if (peerPubKey === nodePubKey) {
+          resolve(rejectionMsg);
+        }
+      });
+    });
+
+    const timeout: Promise<void> = new Promise((resolve) => {
+      setTimeout(resolve, 1000);
```

Comment:
One second seems a little bit short for me. This should be enough most of the time but given a few round trips needed for the handshake and the possibility of intermittent latency, maybe we could do 3 or 5 seconds?

Reply:
@sangaman If the delay is above 2 seconds then it will trigger p2p santity tests' failure

Reply:
Good point, maybe let's just make it 1.5 seconds for now? We can revisit this timeout limit later if it's a problem.